newm-server/src/main/kotlin/io/projectnewm/server/pugins/Sessions.kt

```diff
@@ -0,0 +1,24 @@
+package io.projectnewm.server.pugins
+
+import io.ktor.server.application.Application
+import io.ktor.server.application.install
+import io.ktor.server.sessions.CurrentSession
+import io.ktor.server.sessions.get
+import io.ktor.server.sessions.set
+import io.ktor.server.sessions.Sessions
+import io.ktor.server.sessions.clear
+import io.ktor.server.sessions.cookie
+
+data class UserSession(val token: String)
+
+fun Application.configureSessions() {
+    install(Sessions) {
+        cookie<UserSession>("user-session")
+    }
+}
+
+var CurrentSession.token: String?
+    get() = get<UserSession>()?.token
+    set(value) = set(UserSession(value!!))
```

Comment:
Should we handle the case where `currentSession.token = null`

Reply:
Setting it to null is a programming error